Tests/RegexTests/MatchTests.swift

```diff
@@ -2193,6 +2193,25 @@ extension RegexTests {
       ("cafe", true),
       ("CaFe", true),
       ("EfAc", true))
+    
+    matchTest(
+      #"(?i)a+b"#,
+      ("ab", true),
+      ("Ab", true),
+      ("aB", true),
+      ("AB", true))
+    matchTest(
+      #"(?i)a?b"#,
+      ("ab", true),
+      ("Ab", true),
+      ("aB", true),
+      ("AB", true))
+    matchTest(
+      #"(?i)[a]?b"#,
+      ("ab", true),
+      ("Ab", true),
+      ("aB", true),
+      ("AB", true))
```

Comment:
Could you try a bounded quantification test case?

Reply:
Can you also add test cases for optional characters, which are also broken? See https://github.com/swiftlang/swift/issues/78155